tests/integration/Core/Content/Category/SalesChannel/CategoryRouteTest.php

```diff
@@ -139,11 +162,207 @@ public function testHomeWithSalesChannelOverride(): void
         $this->browser->request(
             'POST',
             '/store-api/category/home',
-            [
-            ]
+            []
+        );
+
+        $this->assertListingCmsPage($this->ids->get('home-category'), $this->ids->get('cms-page'));
+    }
+
+    /**
+     * @return CmsInheritanceDataProviderIterator
+     */
+    public static function categoryCmsPageInheritanceDataProvider(): iterable
+    {
+        yield from CategoryRouteInheritanceFixtures::noOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::mixedOverridesDataProvider();
+        yield from CategoryRouteInheritanceFixtures::allOverridesDataProvider();
+
+        yield from CategoryRouteInheritanceFixtures::enOverridesDataProvider();
```

Comment:
are these 3 override checks really required? I mean the logic is the same, only the languages are different.

Reply:
I can remove some of the tests if you think it's too much, but it's not exactly the same, since en would fall back to Template immediately, while de has en in between and so on. I guess I'd rather shrink the amount than leave a complete "category" of cases out of the tests.